src/newcharacter.cpp

```diff
@@ -1596,11 +1596,11 @@ tab_direction set_profession( avatar &u, points_left &points,
             for( const trait_id &old_trait : u.prof->get_locked_traits() ) {
                 u.toggle_trait( old_trait );
             }
+            const int netPointCost = sorted_profs[cur_id]->point_cost() - u.prof->point_cost();
```

Comment:
I don't know how I managed to do that, so I can't tell you specific reproduce steps yet, but `sorted_profs` was empty for me once and app crashed on this line trying to access `point_cost()`.

Maybe we better guard this with `!sorted_profs.empty()` (and fallback to 0 if it is empty)?

Reply:
I think this can happen when the profession list is empty due to filtering and the screen is refreshed before the "no result" popup is closed. I'll look into it.

Reply:
Could be a profession blacklist thing probably (DSA mode [has them](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/mods/Dark-Skies-Above/overrides/scenarios.json)) - after you switch mods loaded data should be cleared, but it is not always the case (#30242). Maybe profession blacklists are not restored to normal after you start with vanilla, after you've started with the mod, so profession list becomes empty.

Reply:
`DynamicDataLoader::unload_data` already calls `scenario::reset` though, which calls `generic_factory<scenario>::reset`, so I'm not sure how that would happen.